### PR TITLE
Set objectId into query for Email Validation

### LINKED ISSUE #3850: Email verification does not trigger LiveQuery event
### Issue Description

I'm trying to lock users out until they verify their email. I would like to use LiveQuery so I can give user access to the rest of the app once they verify. The problem is that the email verification link does not trigger the LiveQuery update(or enter) event. It also doesn't update the `updatedAt` key on the user object.

If I use the Parse Dashboard to manually change the `emailVerified` key, the LiveQuery update event gets triggered and the `updatedAt` key also gets updated.

### Steps to reproduce

```
let query = new Parse.Query(Parse.User);
query.equalTo('objectId', <user objectId>)
query.equalTo('emailVerified', true)
const subscription = query.subscribe();

subscription.on('enter', (object) => {
        console.log('updated');
	subscription.unsubscribe();
});

subscription.on('open', () => {
	console.log('opened');
});

subscription.on('close', () => {
	console.log('closed');
});
```

#### Expected Results

The LiveQuery update/enter event to be triggered when the user verifies their email

#### Actual Outcome

The LiveQuery update/enter events are not triggered

### Environment Setup

- **Server**
  - parse-server version (Be specific! Don't say 'latest'.) : 2.4.0
  - Operating System:     OS X 10.11.6
  - Hardware:             Macbook Pro 2010
  - Localhost or remote server? (AWS, Heroku, Azure, Digital Ocean, etc): Localhost

- **Database**
  - MongoDB version: 3.2.10
  - Storage engine:  wiredTiger
  - Hardware:       Macbook Pro 2010
  - Localhost or remote server? (AWS, mLab, ObjectRocket, Digital Ocean, etc): Localhost

### LINKED ISSUE #6921: LiveQuery _User and Email Verification
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description
Hello,

We have a problem on Email Validation.

In our client, we listen a liveQuery on _User to detect emailVerified value changes.

But, when we use email verification link, the value change but liveQuery doesn't receive the event.

Same problem as here: #3850

### Actual Outcome

The LiveQuery update/enter events are not triggered

### Expected Outcome
The LiveQuery update/enter event to be triggered when the user verifies their email

###  Environment
Server
- Parse Server version: `4.3.0`
- Operating system: `Linux`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `Local and Sloppy.io host`

Database
- System (MongoDB or Postgres): `MongoDB`
- Database version: `3.6.19-6`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `Local and Sloppy.io`

Client
- SDK (iOS, Android, JavaScript, PHP, Unity, etc): `Flutter`
- SDK version: `1.0.27`

### PR DESCRIPTION
**Pull request by Danaru87**
Fixes #6921

It seems that LiveQueryController need original object id to send event.
But, email validation was not retrieving the user validating email address.

### CODE REVIEW

**Comment by davimacedo** [comment]
Thanks for the PR. Could you please run `npm run lint` and fix the lint issue? It would be also good to add a regression test.

**Comment by Danaru87** [comment]
Lint ok !

For regression test, i have no idea how to do that, can you give me some help ?

**Comment by davimacedo** [comment]
Take a look at this file: https://github.com/parse-community/parse-server/blob/master/spec/ParseLiveQuery.spec.js

You need to create a new test case similar to these ones that reproduces the problem that you are trying to solve. This test case should not pass before your change and pass after your change.

**Comment by Danaru87** [comment]
> Take a look at this file: https://github.com/parse-community/parse-server/blob/master/spec/ParseLiveQuery.spec.js
> 
> You need to create a new test case similar to these ones that reproduces the problem that you are trying to solve. This test case should not pass before your change and pass after your change.

Working on it ;)

**Comment by Danaru87** [comment]
Test done !

File: src/Controllers/UserController.js [hunk 1]
```diff
@@ -64,16 +64,17 @@ export class UserController extends AdaptableController {
...
-      { username: username, emailVerified: true }
+      { username: username }
     );
-    return checkIfAlreadyVerified.execute().then(result => {
-      if (result.results.length) {
+    return findUserForEmailVerification.execute().then(result => {
+      if (result.results.length && result.results[0].emailVerified) {
         return Promise.resolve(result.results.length[0]);
       }
+      query.objectId = result.results[0].objectId;
```
Comment (davimacedo) [inline src/Controllers/UserController.js:77]: Maybe we should check if the reason continues to exist?

```js
if (result.results.length && result.results[0].emailVerified) {
  return Promise.resolve(result.results.length[0]);
} else if (result.results.length) {
  query.objectId = result.results[0].objectId;
}
```

**Review by davimacedo — CHANGES_REQUESTED** [review]
Good job! Only a small suggestion.

File: src/Controllers/UserController.js (hunk 1 above)
Comment (Danaru87) [reply src/Controllers/UserController.js:77]: that's rigth ! I change this ;)

Comment (Danaru87) [reply src/Controllers/UserController.js]: Done ;)

**Review by davimacedo — APPROVED** [review]
LGTM!

**Pull request merged**